tests/Fixer/Whitespace/BlankLineBeforeControlStatementFixerTest.php

```diff
@@ -0,0 +1,1046 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Whitespace;
+
+use PhpCsFixer\Test\AbstractFixerTestCase;
+use PhpCsFixer\WhitespacesFixerConfig;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ */
+final class BlankLineBeforeControlStatementFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider providerInvalidControlStatement
+     *
+     * @param mixed $controlStatement
+     */
+    public function testConfigureRejectsInvalidControlStatement($controlStatement)
+    {
+        $controlStatements = array(
+            'break',
+            'continue',
+            'declare',
+            'do',
+            'else',
+            'elseif',
+            'for',
+            'foreach',
+            'if',
+            'include',
+            'include_once',
+            'require',
+            'require_once',
+            'return',
+            'switch',
+            'throw',
+            'try',
+            'while',
+        );
+
+        $this->setExpectedException(
+            'PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException',
+            sprintf(
+                'Expected one of "%s", got "%s".',
+                implode('", "', $controlStatements),
+                is_object($controlStatement) ? get_class($controlStatement) : gettype($controlStatement)
+            )
+        );
+
+        $this->fixer->configure(array(
+            $controlStatement,
+        ));
+    }
+
+    /**
+     * @return array
+     */
+    public function providerInvalidControlStatement()
+    {
+        return array(
+            'null' => array(null),
+            'false' => array(false),
+            'true' => array(true),
+            'int' => array(0),
+            'float' => array(3.14),
+            'array' => array(array()),
+            'object' => array(new \stdClass()),
+            'unknown' => array('foo'),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithReturn
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithDefaultConfiguration($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @dataProvider providerFixWithBreak
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithBreak($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'break',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithBreak()
+    {
+        return array(
+            array(
+'<?php
+switch ($a) {
+    case 42:
+        break;
+}',
+            ),
+            array(
+'<?php
+switch ($a) {
+    case 42:
+        $foo = $bar;
+
+        break;
```

Comment:
When inside a switch, I also remove any blank lines between `break` and next `case`/`default`, should this fixer do that too?